AR-js-org/AR.js
# Add option to use initial location as world origin

**⚠️ All PRs have to be done versus 'dev' branch, so be aware of that, or we'll close your issue ⚠️**

**What kind of change does this PR introduce?**


Adds the facility to use the initial GPS location as the WebGL/three.js/A-Frame world origin, as discussed in issue #499.

**Can it be referenced to an Issue? If so what is the issue # ?**

#499 

**How can we test it?**



A new example `initial-location-as-origin` has been added to the A-Frame `new-location-based` examples. Try this, either on mobile or on desktop (with the latter, you will have to manually enter a latitude and longitude if your location cannot be determined). Click on any of the four visible boxes. These will show each box's world coordinates and either the `x` or `z` should have a value less than 200. Furthermore the camera position should be reported as x=0, y=1.6, z=0.

**Summary**


This is done via a new `initialPositionAsOrigin` property of `gps-new-camera`. It's `false` by default, so as not to break existing applications, but recommended to be set to `true` for new code.

The rationale for this is discussed in issue #499, namely that very large absolute values for world coordinates can cause precision problems. By setting the world origin to the initial GPS position (as was the case in the older location based components, as it happens) this problem is lessened.

**Does this PR introduce a breaking change?**

No - original behaviour is maintained if the property is not set.

**Please TEST your PR before proposing it. Specify here what device you have used for tests, version of OS and version of Browser**

Tested with Firefox on desktop (Linux) and Android 12/Chrome latest (Pixel 3), the latter using a real GPS location.


**Other information**

My proposal is to make this the default at version 3.5, but not yet to avoid breaking changes. Also at 3.5 the floating origin feature of #499 will be introduced, I think this will require a more significant rewrite.
